src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -110,9 +111,21 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	private Integer getNumPartitions(String topic) {
+		try {
+			return this.kafka.partitionsFor(topic).size(); //returns 1 for new topics
+		} catch(TimeoutException e) {
+			LOGGER.error("Topic '" + topic + "' name does not exist. Exception: " + e.getLocalizedMessage());
+			throw e;
```

Comment:
looked into this; it's basically sucky.  for both 0.8 and 0.9 producers, with auto-creation disabled, an attempt to call `partitionsFor()` on a non-existent topic just results in a `TimeoutException` the same way that Sarah coded for.  In 0.9 you get 

```
13:16:41,371 WARN  NetworkClient - Error while fetching metadata with correlation id 563 : {foijlkj=UNKNOWN_TOPIC_OR_PARTITION}
```

logged, but nothing programattic. 

There's also the `listTopics()` method on the consumer; maybe that would be a more elegant way to do this.

Really what I'd like is for maxwell to crash cleanly if you attempt to produce to a non-existent topic.  The `TimeoutException` method is... sorta viable... but I'll ask on the kafka mailing list.